lib/Doctrine/Common/Annotations/FileCacheReader.php

```diff
@@ -159,6 +159,41 @@ public function getPropertyAnnotations(\ReflectionProperty $property)
         return $this->loadedAnnotations[$key] = include $path;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function getConstantAnnotations(\ReflectionClassConstant $constant)
+    {
+        $class = $constant->getDeclaringClass();
+        if ( ! isset($this->classNameHashes[$class->name])) {
+            $this->classNameHashes[$class->name] = sha1($class->name);
+        }
+        $key = $this->classNameHashes[$class->name].'::'.$constant->getName();
+
+        if (isset($this->loadedAnnotations[$key])) {
+            return $this->loadedAnnotations[$key];
+        }
+
+        $path = $this->dir.'/'.strtr($key, '\\', '-').'.cache.php';
+        if (!is_file($path)) {
+            $annot = $this->reader->getConstantAnnotations($constant);
+            $this->saveCacheFile($path, $annot);
+            return $this->loadedAnnotations[$key] = $annot;
+        }
+
+        if ($this->debug
+            && (false !== $filename = $class->getFilename())
+            && filemtime($path) < filemtime($filename)) {
+            @unlink($path);
+
+            $annot = $this->reader->getConstantAnnotations($constant);
+            $this->saveCacheFile($path, $annot);
+            return $this->loadedAnnotations[$key] = $annot;
+        }
+
+        return $this->loadedAnnotations[$key] = include $path;
+    }
```

Comment:
There seems to be a lot of copy/paste in this file, does anybody know why?

Reply:
No idea. But it is beeing removed in 2.0 anyway.

Reply:
Ok, this might be fine then :)